### PR TITLE
Improve server side analytics

### PR DESCRIPTION
**Pull request by phillipthelen**
### Changes

This PR changes two things.
1. Added the following information to server side amplitude events:
   - set platform to "Web", "iOS", "Android" or "3rd Party" depending on the client that sent the request
   - Set "OS" to either the browser or the mobile OS
2. Moved the following events from the mobile client to the server
   - login
   - Creating a guild/party
   - joining a guild/party

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Since the headers are passed into every session, would it make more sense to pass the `req.headers` in and let the analytics service pull out the data?

**Comment by paglias** [comment]
I agree with @crookedneighbor 's comment

File: common/script/ops/rebirth.js [hunk 1]
```diff
@@ -30,6 +30,8 @@ module.exports = function rebirth (user, tasks = [], req = {}, analytics) {
   }
 
   if (analytics) {
+    analyticsData.client = req.headers['x-client'];
```
Comment (crookedneighbor) [inline common/script/ops/rebirth.js]: Any reason this isn't just passing in headers?

File: website/server/controllers/api-v3/groups.js [hunk 2]
```diff
@@ -66,6 +66,30 @@ api.createGroup = {
       _id: user._id,
       profile: {name: user.profile.name},
     };
+
+    if (savedGroup.privacy === 'public') {
```
Comment (crookedneighbor) [inline website/server/controllers/api-v3/groups.js:80]: How about 

``` js
let analyticsObject = {
  uuid: user._id,
  hitType: 'event',
  category: 'behavior',
  owner: true,
  groupType: savedGroup.type,
  privacy: savedGroup.privacy,
  headers: req.headers
};

if (savedGroup.privacy === 'public') {
  analyticsObject.groupName = savedGroup.name;
}

res.analytics.track('join group', analyticsObject);
```

File: website/server/controllers/api-v3/groups.js [hunk 3]
```diff
@@ -273,6 +297,30 @@ api.joinGroup = {
     if (leader) {
       response.leader = leader.toJSON({minimize: true});
     }
+
+    if (group.privacy === 'public') {
```
Comment (crookedneighbor) [inline website/server/controllers/api-v3/groups.js:304]: Same here.

File: website/server/libs/analyticsService.js [hunk 4]
```diff
@@ -81,13 +82,45 @@ let _formatUserData = (user) => {
   return properties;
 };
 
+let _formatPlatformForAmplitude = (platform) => {
+  switch (platform) {
```
Comment (crookedneighbor) [inline website/server/libs/analyticsService.js]: I'd prefer an if/else if/else logic here. Or a map:

``` js
const PLATFORM_MAP = Object.freeze({
  'habitica-web': 'Web',
  'habitica-ios': 'iOS',
  'habitica-android': 'Android',
});

function _formatPlatformForAmplitude (platform) {
  if (platform in PLATFORM_MAP) {
    return PLATFORM_MAP[platform];
  }

  return platform;
}

```

File: website/server/libs/analyticsService.js [hunk 5]
```diff
@@ -100,10 +133,10 @@ let _formatDataForAmplitude = (data) => {
   if (itemName) {
     event_properties.itemName = itemName;
   }
-
   return ampData;
 };
 
+
```
Comment (crookedneighbor) [inline website/server/libs/analyticsService.js]: Extra line break here.

**Comment by phillipthelen** [comment]
made the requested changes

**Pull request closed without merging**